### PR TITLE
Added Ranking Based on Stocks You Follow

### PR DESCRIPTION
**Pull request by javidangarcia**
- Added ranking component and user interface for seeing ranked stocks.
- Added /ranking endpoint that ranks 10 stocks based on stocks the user follows. If the user follows less than 10 stocks, then the algorithm chooses different stocks at random that are in the database for the rest of the ranking.

Note: This is a basic implementation and will become more complex when likes, dislikes, and friends are added.

https://github.com/javidangarcia/stock-zone/assets/89328228/8ffeef73-ac0b-4fb5-a09f-2fa73bc5bb1a

### CODE REVIEW

File: backend/routes/follows.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ router.post("/follow", async (req, res) => {
     const user = req.session.user;
     const ticker = req.body.ticker?.toUpperCase();
 
-    if (!ticker) {
+    if (ticker == null) {
```
Comment (vandanasub) [inline backend/routes/follows.js:45]: I like the added explicit null checking here

File: backend/routes/follows.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ router.post("/follow", async (req, res) => {
 
         const follow = await Follow.findOne({ where: followData });
 
-        if (follow != null) {
+        if (follow !== null) {
```
Comment (vandanasub) [inline backend/routes/follows.js:65]: Curious, is there a reason you've changed it this way? It this never undefined?

File: backend/routes/rankings.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            do {
+                const randomIndex = Math.floor(
+                    Math.random() * stocksInDatabase.length
+                );
+                potentialStock = stocksInDatabase[randomIndex];
+
+                alreadyInRanking = stocksRanking.some(
+                    (stock) => stock.name === potentialStock.name
+                );
+            } while (alreadyInRanking);
```
Comment (vandanasub) [inline backend/routes/rankings.js:54]: this kind of logic could be pulled out in it's own helper function or add a comment. You're essentially trying to pick `randomStocksNeeded` number of elements from the stocks database. You could try [googling](https://bobbyhadz.com/blog/javascript-get-multiple-random-elements-from-array) this to see if there's a cleaner way to do something similar with `slice` or `sort`

**Review by vandanasub — COMMENTED** [review]
Nice job!

File: backend/routes/follows.js (hunk 2 above)
Comment (javidangarcia) [reply backend/routes/follows.js:65]: Should I put it to include undefined as well? The findOne returns an empty array if nothing is found.

File: backend/routes/follows.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ const router = express.Router();
 
 const checkSession = (req, res, next) => {
     const user = req.session.user;
-    if (!user) {
+    if (user == null) {
```
Comment (BriptimusPrimus) [inline backend/routes/follows.js:9]: 👍

File: backend/routes/follows.js (hunk 1 above)
Comment (BriptimusPrimus) [reply backend/routes/follows.js:45]: Me too :D

File: backend/routes/rankings.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import express from "express";
+import { Follow } from "../models/follow.js";
+import { Stock } from "../models/stock.js";
+
+const router = express.Router();
+
+const checkSession = (req, res, next) => {
```
Comment (BriptimusPrimus) [inline backend/routes/rankings.js:7]: No need to repeat this function everywhere. Let's have a common one, and then import it and "use it" `router.use()` where needed.

File: frontend/src/components/StockNews/StockNews.jsx [hunk 6]
```diff
@@ -38,13 +38,9 @@ export default function StockNews({ stocks }) {
...
     const handleOptions = (event) => {
         const ticker = event.target.value;
-        stocks.map((stock) => {
-            if (stock.ticker === ticker) {
-                setCurrentStock(stock);
-                return;
-            }
-        })
-    }
+        const selectedStock = stocks.find((stock) => stock.ticker === ticker);
```
[the remark this answers is not in the document]
Comment (BriptimusPrimus) [reply frontend/src/components/StockNews/StockNews.jsx:41]: 👍

**Pull request merged**